### ISSUE TITLE
L001 and L008: Indented dbt_utils jinja macro leads to unwanted unnecessary trailing whitespace error

### ISSUE THREAD

**Issue by jamesleemm** [sqlfluff/sqlfluff#4475]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

(Note: This issue seems superficially similar to a [closed issue 3988](https://github.com/sqlfluff/sqlfluff/issues/3988), but I believe this has a different root cause.)

My `gold_bi_person_daily.sql` model contains an indented dbt_utils jinja macro

```
{{
  config(
    alias = 'person_daily'
    )
}}

select
    date,
    {{ dbt_utils.generate_surrogate_key('date', 'harvest_user_id') }} as id
from {{ ref('silver_operations_project_revenue_daily') }}
```

This gets compiled as

```
select
    date,
    
    
md5(cast(coalesce(cast(date as TEXT), '_dbt_utils_surrogate_key_null_') || '-' || coalesce(cast(harvest_user_id as TEXT), '_dbt_utils_surrogate_key_null_') as TEXT)) as id
from DBT_OPS_REPORTING_DEV_james.silver_operations.project_revenue_daily
```

And the following error is returned
```
== [models/gold/bi/gold_bi_person_daily.sql] FAIL                                                               
L:   9 | P:   1 | L001 | Unnecessary trailing whitespace.
L:   9 | P:   1 | L008 | Unnecessary trailing whitespace.
```

These errors are not returned if the dbt_utils macro is not indented.

What I believe is happening for the L001 error is that the macro starts with a \n newline. This means that the indented
`    {{ dbt_utils.generate_surrogate_key('date', 'harvest_user_id') }} as id`
gets compiled as
`    \n {{ dbt_utils.generate_surrogate_key('date', 'harvest_user_id') }} as id`
leaving a line that contains only spaces.

I believe the solution here would be to ignore leading indents on templated jinja that end up on a new line.

I'm not sure what is causing the L008 error.

### Expected Behaviour

There should be no error.

### Observed Behaviour

```
== [models/gold/bi/gold_bi_person_daily.sql] FAIL                                                               
L:   9 | P:   1 | L001 | Unnecessary trailing whitespace.
L:   9 | P:   1 | L008 | Unnecessary trailing whitespace.
```

### How to reproduce

.sql model provided above

### Dialect

snowflake

### Version

sqlfluff, version 1.4.5
Python 3.10.7
sqlfluff-templater-dbt

### Configuration

`.sqlfluff`
```
[sqlfluff]
templater = dbt
dialect = snowflake
# level of log output
verbose = 0
# run maximum parallel processes
processes = 0
# Comma separated list of rules to check
rules = all
# Comma separated list of rules to exclude
exclude_rules = L031, L061
warnings = L034

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-indent-locations
indented_on_contents = false
template_blocks_indent = true

# Layout configuration
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-layout-and-spacing
[sqlfluff:layout:type:comment]
spacing_before = single
spacing_after = touch

# Some rules can be configured directly from the config common to other rules
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space
allow_scalar = False
single_table_references = consistent
unquoted_identifiers_policy = all

[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = none
ignore_words_regex = none
unquoted_identifiers_policy = all

[sqlfluff:rules:L029]
# Keywords should not be used as identifiers.
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = aliases
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L030]
# Function names
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L040]
# Null & Boolean Literals
capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L062]
# Comma separated list of blocked words that should not be used
# Add "-- noqa: L062" to the end of the line for exceptions in the code base that should be allowed
blocked_words = distinct, right, full
blocked_regex = None

[sqlfluff:rules:L063]
# Data Types
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None
```

`.sqlfluffignore`
```
# Files that should not be linted.

target/

# dbt <1.0.0
dbt_modules/ 

# dbt >=1.0.0
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
I _think_ this might be a duplicate of #4485 and therefore fixed in #4491. I'll confirm that before closing however.

**Comment by alanmcruickshank** [comment]
I've managed to confirm this on `1.4.5`, but it is no longer present on `2.0.0a5` and `2.0.0a6`. I'm going to close this as resolved. `2.0.0` will be released soon which will contain the fixes.
